src/Symfony/Component/Form/Extension/DataCollector/FormDataCollector.php

```diff
@@ -238,7 +238,7 @@ public function serialize()
             }
         }
 
-        return serialize($this->cloneVar($this->data));
+        return serialize($this->cloneVar($this->data)->getValue(true));
     }
```

Comment:
Can you please confirm that it works on e.g. the demo app? Can you also please push force again to trigger the CI so that tests run?

Reply:
It's not good on the demo, I still have to fix the method `WebProfilerExtension::dumpData()` which asks for an instance of Data

Reply:
I think this change is wrong actually - we don't want to resolve the `Data` before serializing it

Reply:
Instead, maybe try calling the method (without any argument) inside `getData`?
please also check how the other data collector work; is this one the only one that stores a `Data` object in the data property?

Reply:
It does not work either without any argument.
After some research I noticed on all the other `DataCollector` the `data` type never changes but the values ​​that are dump are.
This solution works in demo with the same behavior as before.
What do you think of that ?